.github/workflows/dependabot-auto-merge.yml

```diff
@@ -0,0 +1,39 @@
+name: Dependabot auto-merge
+on:
+  workflow_call:
+    inputs:
+      merge-option:
+        description: 'Merge option'
+        type: string
+        required: false
+        default: '--squash'
+    secrets:
+      token:
+        description: 'Token used to enable auto-merge and auto-approve'
+        required: true
+
+permissions:
+  pull-requests: write
+  contents: write
+
+env:
+  PR_URL: ${{ github.event.pull_request.html_url }}
+  # this token CANNOT be the default `GITHUB_TOKEN` on auto-merge action, otherwise the merge of the PR will not trigger a build
+  GITHUB_TOKEN: ${{ secrets.token }}
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.user.login == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.3.3
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
Shouldn't this be `${{ env.GITHUB_TOKEN }}`?

Reply:
My understanding looking at the code is that using the secrets context is the correct way of referencing this token:https://docs.github.com/en/actions/security-guides/automatic-token-authentication

Reply:
(and this has been tested successfully without the secrets: inherit setting from the caller)